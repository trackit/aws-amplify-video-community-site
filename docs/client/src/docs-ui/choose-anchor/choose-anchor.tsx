import {Component, Host, h, Prop} from "@stencil/core";
import {Page} from "../../api";
import {getFilterKeyFromPage} from "../../utils/filters";
import {filterMetadataByOptionByName} from "../../utils/filter-data";

@Component({tag: "docs-choose-anchor", shadow: false})
export class DocsChooseAnchor {
  /*** the current page's data */
  @Prop() readonly page: Page;

  render() {
    const filterKey = getFilterKeyFromPage(this.page);

    return (
      <Host>
        <amplify-responsive-grid
          gridGap={1}
          columns={4}
          class="border-radius margin-top-md"
        >
          {filterKey &&
            Object.entries(filterMetadataByOptionByName[filterKey]).map(
              ([filterValue, {label, graphicURI}]) => {
                const route = this.page.versions?.[filterValue] as string;

                return (
                  <docs-card key={label} vertical url={route}>
                    <img
                      slot="graphic"
                      src={graphicURI}
                      alt={`${label} Logo`}
                    />
                    <h4 slot="heading">{label}</h4>
                  </docs-card>
                );
              },
            )}
        </amplify-responsive-grid>
      </Host>
    );
  }
}
