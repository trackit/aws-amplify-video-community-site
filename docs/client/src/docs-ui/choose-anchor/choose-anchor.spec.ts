import {DocsChooseAnchor} from "./choose-anchor";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-choose-anchor", () => {
  describe("Component logic", () => {
    let chooseAnchor: DocsChooseAnchor;
    beforeEach(() => (chooseAnchor = new DocsChooseAnchor()));

    it("should init `page` as `undefined`", () =>
      expect(chooseAnchor.page).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsChooseAnchor],
            html: `<docs-choose-anchor />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
