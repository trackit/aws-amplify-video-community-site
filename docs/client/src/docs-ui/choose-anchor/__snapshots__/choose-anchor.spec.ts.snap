// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-choose-anchor Render logic should render 1`] = `
<docs-choose-anchor>
  <amplify-responsive-grid class="border-radius margin-top-md" columns="4" gridgap="1"></amplify-responsive-grid>
</docs-choose-anchor>
`;
